subsys/bluetooth/common/Kconfig

```diff
@@ -59,6 +59,13 @@ config BT_RPA
 	select TINYCRYPT
 	select TINYCRYPT_AES
 
+config BT_PROPRIETARY_EXT
```

Comment:
We already have:

```
BT_HAS_HCI_VS
BT_HCI_VS
BT_HCI_VS_EXT
BT_HCI_VS_EXT_DETECT
```
plus then the ones that are being discussed as part of https://github.com/zephyrproject-rtos/zephyr/pull/16106

Can we discuss the scope and settle here?

Reply:
This is not related to HCI per se, it is a more generic hook

Reply:
Then it should not be in `subsys/bluetooth/common/Kconfig`? It should be in `subsys/bluetooth/controller/Kconfig` ?

Reply:
I did discuss this with Thomas a few days ago, the idea is that this could also be used to enable proprietary extensions in the host, as both host and ctrl  implementations are likely needed

Reply:
Looking in my crystal ball i was thinking that we might also need some non-HCI proprietary hooks in the host and thats' why i suggested the current location to Asger.

It might have been a premature decision from my side, so the solution would be to move the Kconfig back to the `subsys/bluetooth/controller/Kconfig` and when we have a need for it in the host we will then move it to `subsys/bluetooth/common/Kconfig` at that point.

Reply:
I might be tempted to call this `BT_CTLR_USER_EXT` now that we've started using the word "user" for this sort of stuff and this belongs in the controller. If we need additional extensions in the Host we can deal with those later.

Reply:
I changed the constant name to BT_CTLR_USER_EXT  and moved it to subsys/bluetooth/controller/Kconfig